web/src/hooks/useSearch.ts

```diff
@@ -74,11 +76,12 @@ export const extractAnswer = async (
 
       case "streaming":
         // incrementally build up the response
+        const content = formatCitations((result?.content || "") + data.content);
         result = {
+          content,
           role: "assistant",
-          content: (result?.content || "") + data.content,
           citations: result?.citations || [],
-          base_count: result?.base_count || baseReferencesIndex,
+          citationsMap: findCitations(content, result?.citations || []),
```

Comment:
The LLM returns citations like "bla bla bla [a] ble ble [b][c]". These then need to be mapped to the appropriate numbers. This `citationMap` is a mapping of `{<letter>: <full citation object>}`. It's possible to change from letters to numbers - it's all in the LLM prompt